packages/asset/contracts/AssetCreate.sol

```diff
@@ -0,0 +1,351 @@
+    /// @dev Called from the bridge contract
+    /// @param creator The address of the creator
+    /// @return nonce The next available creator nonce
+    function bridgeGetCreatorNonce(
+        address creator
+    ) external onlyRole(BRIDGE_MINTER_ROLE) returns (uint16) {
+        creatorNonces[creator]++;
+        return creatorNonces[creator];
+    }
+
+    /// @notice Get the asset contract address
+    /// @return The asset contract address
+    function getAssetContract() external view returns (address) {
+        return address(assetContract);
+    }
+
+    /// @notice Get the catalyst contract address
+    /// @return The catalyst contract address
+    function getCatalystContract() external view returns (address) {
+        return address(catalystContract);
+    }
+
+    /// @notice Get the auth validator address
+    /// @return The auth validator address
+    function getAuthValidator() external view returns (address) {
+        return address(authValidator);
+    }
+
+    /// @notice Creates a hash of the mint data
+    /// @param creator The address of the creator
+    /// @param tier The tier of the asset
+    /// @param amount The amount of copies to mint
+    /// @param metadataHash The metadata hash of the asset
+    /// @return digest The hash of the mint data
+    function _hashMint(
+        address creator,
+        uint16 nonce,
+        uint8 tier,
+        uint256 amount,
+        bool revealed,
+        string calldata metadataHash
+    ) internal view returns (bytes32 digest) {
+        digest = _hashTypedDataV4(
+            keccak256(
+                abi.encode(
+                    MINT_TYPEHASH,
+                    creator,
+                    nonce,
+                    tier,
+                    amount,
+                    revealed,
+                    keccak256((abi.encodePacked(metadataHash)))
+                )
+            )
+        );
+    }
+
+    /// @notice Creates a hash of the mint batch data
+    /// @param creator The address of the creator
+    /// @param tiers The tiers of the assets
+    /// @param amounts The amounts of copies to mint
+    /// @param metadataHashes The metadata hashes of the assets
+    /// @return digest The hash of the mint batch data
+    function _hashBatchMint(
+        address creator,
+        uint16 nonce,
+        uint8[] calldata tiers,
+        uint256[] calldata amounts,
+        bool[] calldata revealed,
+        string[] calldata metadataHashes
+    ) internal view returns (bytes32 digest) {
+        digest = _hashTypedDataV4(
+            keccak256(
+                abi.encode(
+                    MINT_BATCH_TYPEHASH,
+                    creator,
+                    nonce,
+                    keccak256(abi.encodePacked(tiers)),
+                    keccak256(abi.encodePacked(amounts)),
+                    keccak256(abi.encodePacked(revealed)),
+                    _encodeHashes(metadataHashes)
+                )
```

Comment:
Note: check if enough tests for collisions for tiers / amounts / revealed, since we use encodePacked

Reply:
If tiers, amounts, revealed are passed in any other order the signature should not work